Zend/zend_vm_def.h

```diff
@@ -6047,23 +6047,39 @@ ZEND_VM_C_LABEL(add_unpack_again):
 	if (EXPECTED(Z_TYPE_P(op1) == IS_ARRAY)) {
 		HashTable *ht = Z_ARRVAL_P(op1);
 		zval *val;
-		zend_string *key;
+		
+		if (HT_IS_PACKED(ht) && (zend_hash_num_elements(result_ht) == 0 || HT_IS_PACKED(result_ht))) {
```

Comment:
```suggestion
		if (HT_IS_PACKED(ht) && (!HT_IS_INITIALIZED(result_ht) || HT_IS_PACKED(result_ht))) {
```

I'm pretty sure there can be hash tables that are initialized but associative(not packed), e.g. `$globalXyz = ['a' => 'val']; unset($globalXyz['a']);` has zend_hash_num_elements === 0

Reply:
Yeah this is just to check if the temp array needs to be mark as packed and/or extend to necessary size. `zend_hash_num_elements` is for new array.

I'll do a benchmark later to see there is any performance improvement, although I won't expect much difference. `zend_hash_num_elements` is inline and simple.

Reply:
> since both macros are checking HT_FLAGS from the same location in memory and the compiler could optimize that

And following up, unsurprisingly, the compiler does optimize that (one less branch to predict, so I'd expect it to be better, though for it to be that noticeable you'd use a random mix of lists of length 0 and 1)

https://godbolt.org/z/sdaPEf8z9

```c
void xyz(int n) {
    if ((n & (1<<2)) || !(n&(1<<3))) {
        puts("Matches one flag and not the other");
    }
}
```

e.g. the condition compiles to `(x&12) != 8`

```asm
        and     edi, 12
        cmp     edi, 8
        jne     .LBB0_2
```